File: src/Mesh.js
const MeshType = require('./MeshType.js')
const bufferToTypedArray = require('./bufferToTypedArray.js')

const Mesh = function (meshType = new MeshType()) {
  this.meshType = meshType

  this.name = 'Mesh'

  this.numberOfPoints = 0
  this.points = bufferToTypedArray(meshType.pointComponentType, new ArrayBuffer(0))

  this.numberOfPointPixels = 0
  this.pointData = bufferToTypedArray(meshType.pointPixelComponentType, new ArrayBuffer(0))

  this.numberOfCells = 0
  this.cells = bufferToTypedArray(meshType.cellComponentType, new ArrayBuffer(0))

  this.numberOfCellPixels = 0
  this.cellData = bufferToTypedArray(meshType.cellPixelComponentType, new ArrayBuffer(0))
  this.cellBufferSize = 0
}

module.exports = Mesh

File: packages/core/python/itkwasm/itkwasm/mesh.py
from dataclasses import dataclass, field

from typing import Optional, Union, Dict

try:
    from numpy.typing import ArrayLike
except ImportError:
    from numpy import ndarray as ArrayLike

from .float_types import FloatTypes
from .int_types import IntTypes
from .pixel_types import PixelTypes

@dataclass
class MeshType:
    dimension: int = 3

    pointComponentType: Union[IntTypes, FloatTypes] = FloatTypes.Float32
    pointPixelComponentType: Union[IntTypes, FloatTypes] = FloatTypes.Float32
    pointPixelType: PixelTypes = PixelTypes.Scalar
    pointPixelComponents: int = 1

    cellComponentType: Union[IntTypes, FloatTypes] = IntTypes.Int32
    cellPixelComponentType: Union[IntTypes, FloatTypes] = FloatTypes.Float32
    cellPixelType: PixelTypes = PixelTypes.Scalar
    cellPixelComponents: int = 1


@dataclass
class Mesh:
    meshType: Union[MeshType, Dict] = field(default_factory=MeshType)

    name: str = 'Mesh'

    numberOfPoints: int = 0
    points: Optional[ArrayLike] = None

    numberOfPointPixels: int = 0
    pointData: Optional[ArrayLike] = None

    numberOfCells: int = 0
    cells: Optional[ArrayLike] = None
    cellBufferSize: int = 0

    numberOfCellPixels: int = 0
    cellData: Optional[ArrayLike] = None

    def __post_init__(self):
        if isinstance(self.meshType, dict):
            self.meshType = MeshType(**self.meshType)
